coderd/database/queries/workspaces.sql

```diff
@@ -184,6 +194,40 @@ WHERE
 			workspaces.owner_id = @owner_id
 		ELSE true
 	END
+	-- Filter by build parameter
+   	-- @has_param will match any build that includes the parameter.
+	AND CASE WHEN array_length(@has_param :: text[], 1) > 0  THEN
```

Comment:
As this operates on 1 and 0, maybe we can simplify it:

```sql
AND (
    array_length(@has_param :: text[], 1) = 0 
    OR EXISTS (
        SELECT 1
        FROM workspace_build_parameters
        WHERE 
            workspace_build_parameters.workspace_build_id = latest_build.id AND
            workspace_build_parameters.name ILIKE ANY(@has_param)
    )
)
```

Reply:
that is true, however we have this `CASE WHEN` syntax for other params, and other array style params. Do you feel it's worth having a differing syntax?

Reply:
I admit that I like the consistency, although it is tempting to simplify this query wherever it is possible. Anyway, this is not a must-have, your syntax is also correct 👍

Reply:
I'm going to decide to leave it for consistency. All the params are identical right now in how they are handled.